### ISSUE TITLE
is Server-Side Rendering guide still valid?

### ISSUE THREAD

**Issue by wujekbogdan** [vuejs/vuejs.org#751]
This is a copy-paste of a post that I've written on the [Vue forum](https://forum.vuejs.org/t/vue-js-applications-vs-seo-is-server-side-rendering-really-necessary/5957).

I'm building an application where all the content comes from a REST API. The application must be crawlable. I've just read the [Server-Side rendering guide](https://vuejs.org/v2/guide/ssr.html) and the first sentence says very clearly that if a content is loaded asynchronously it will not be visible for crawlers.

> Google and Bing can index synchronous JavaScript applications just fine. **Synchronous being the key word there**. If your app starts with a loading spinner, then fetches content via Ajax, the crawler will not wait for you to finish.

> This means if you have content fetched asynchronously on pages where SEO is important, SSR might be necessary.

It used to be true in the past couldn't but is it still true?

[Here's a blog post from the official Google blog](https://webmasters.googleblog.com/2015/10/deprecating-our-ajax-crawling-scheme.html). We can read there that:

> (...) Times have changed. Today, as long as you're not blocking Googlebot from crawling your JavaScript or CSS files, **we are generally able to render and understand your web pages like modern browsers**.

and

> (...) In general, **websites shouldn't pre-render pages only for Google** -- we expect that you might pre-render pages for performance benefits for users and that you would follow progressive enhancement guidelines.

So is it really true that pages that load their content via AJAX are not crawlable? According to the post on the Google blog, I would expect that it's no longer an issue.

Could you confirm is it true? If so, then the guide should be updated.

**Comment by chrisvfritz** [comment]
The last time I tested it was in June 2016, well after that post was published by Google. It's possible that they've gotten better since June, but I haven't heard anything to indicate a change in strategy.

**Comment by chrisvfritz** [comment]
@wujekbogdan If you're interested, you could actually test this yourself. To test with Google, simply register your site (or a demo site) with Google Webmaster Tools, then request a crawl. When it's done, you'll be able to see the rendered result. If it's a loading screen, SSR, prerendering, or just a good old-fashioned server-side app will indeed be necessary. If it seems to be working exactly as you'd hope, I'll revisit this and update the guide accordingly!

**Comment by wujekbogdan** [comment]
I'll do some tests and post the results here.

**Comment by wujekbogdan** [comment]
I did some tests with Search Console using the _Fetch by Google_ feature:

- PageS rendered with JS are visible for Google when they're rendered synchronously (as said in the documentation)
- I wasn't able to test if routes are visible for Google if a router is in the hash mode. Google ignores hash and renders a home page instead. It doesn't mean that they are not indexed. It just means that I wasn't able to perform tests.
- Routes in history mode are visible for Google
- When a page content is queried asynchronously (for example from a REST API) then Google Search Console returns an Error status. Still, it doesn't mean that the page is not indexed, but it doesn't mean it's indexed either. But I'm afraid it's not indexed :/

**Comment by chrisvfritz** [comment]
Thanks for double checking this! It sounds like things probably haven't change then. 😔  I'll close this for now, but happy to reopen in the future pending new results.

**Comment by jardakotesovec** [comment]
@chrisvfritz @wujekbogdan Interestingly enough I ran into this recent [post](https://vuejsdevelopers.com/2018/04/09/single-page-app-seo/) testing that async renders works fine.. at least with google bot..  so might worth to reopen?

**Comment by mrcsmcln** [comment]
I know this is only anecdotal, but I launched a Vue Router + Fetch + WordPress REST API site about a month ago and was required to sign up for a prerendering service due to poor SEO. Just prior to launching this site, I did some quick tests using the same technologies and Google crawled it just fine. Not sure if it has to do with the number of async requests, the domain from which I was fetching, or Google was just testing things out, but it was definitely being crawled and that is no longer the case.
